rtsp.go

```diff
@@ -86,13 +99,20 @@ type rtspCamera struct {
 	latestFrame atomic.Pointer[image.Image]
 
 	logger logging.Logger
+
+	rtpH264Passthrough bool
+
+	subsMu       sync.RWMutex
+	subAndCBByID map[rtppassthrough.SubscriptionID]subAndCB
 }
 
 // Close closes the camera. It always returns nil, but because of Close() interface, it needs to return an error.
 func (rc *rtspCamera) Close(ctx context.Context) error {
 	rc.cancelFunc()
+	rc.unsubscribeAll()
+	rc.closeConnection()
 	rc.activeBackgroundWorkers.Wait()
-	return rc.closeConnection()
+	return nil
```

Comment:
We should first close all connections & then wait for all background goroutines to terminate, not vice versa